### DISCUSSION TITLE
Running Gcode and Zeroing

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#161] [Q&A]
Was wondering before I jump into starting to run Gcode if it uses the zeroing we set onto the controller?

Aka is the workflow to take a light cut, then Press A to zero. And similarly to face and zero X.

I understand that it ignores softendstops here so caution is required.

Quite keen to start using the lathecode as I have around 100 parts to make that are identical lol.

Also I noticed when manually moving it doesnt seem to respect the pitch value. It would be handy if once could enforce this.

For example if I just wanted to make a quick cut I could set the Z distance to 10mm then click move and it could cut 10mm but as it stands it would do that movement a bit quickly. If it respected the pitch value we could raise it to speed up or lower it to slow it down.

**Comment by kachurovskiy** [comment]
> Gcode if it uses the zeroing we set onto the controller?

Yes.

> Aka is the workflow to take a light cut, then Press A to zero. And similarly to face and zero X.

Zeroing X to be on the centerline is most important. Zeroing Z is less important, you can just clamp the stock close to the tool each time and take 0.5mm off in lathecode for each part.

> I understand that it ignores softendstops here so caution is required.

Yes, try setting Z=0 away from the stock initially and doing the test cut in the air. Then you can set Z=0 where your stock begins and do the real cut if all goes well.

> Also I noticed when manually moving it doesnt seem to respect the pitch value. It would be handy if once could enforce this.

See https://github.com/kachurovskiy/nanoels/blob/main/h4/README.md#moving-left-and-right-in-and-out

> Depending on the state of the controller, pressing the move button will result in traveling the following distance:
> 
>     off - carriage will move by the step distance but at least 1 stepper motor step
>     ON - carriage will move in pitch increments (stay in the thread) but at least the step distance

> For example if I just wanted to make a quick cut I could set the Z distance to 10mm then click move and it could cut 10mm but as it stands it would do that movement a bit quickly. If it respected the pitch value we could raise it to speed up or lower it to slow it down.

Sorry, don't quite understand. You can [set step to 10mm](https://github.com/kachurovskiy/nanoels/blob/main/h4/README.md#setting-step) - then left/right moves would move you by 10mm each time.
